edb/lib/std/30-datetimefuncs.edgeql

```diff
@@ -56,32 +56,46 @@ std::datetime_get(dt: std::datetime, el: std::str) -> std::float64
 
 
 CREATE FUNCTION
-std::duration_get(dt: std::duration, el: std::str) -> std::float64
+std::datetime_trunc(dt: std::datetime, unit: std::str) -> std::datetime
 {
-    SET volatility := 'IMMUTABLE';
+    # date_trunc of timestamptz is STABLE in PostgreSQL
+    SET volatility := 'STABLE';
     USING SQL $$
-    SELECT date_part("el", "dt")
+    SELECT date_trunc("unit", "dt")
     $$;
 };
 
 
 CREATE FUNCTION
-std::datetime_trunc(dt: std::datetime, unit: std::str) -> std::datetime
+std::duration_trunc(dt: std::duration, unit: std::str) -> std::duration
 {
-    # date_trunc of timestamptz is STABLE in PostgreSQL
-    SET volatility := 'STABLE';
+    SET volatility := 'IMMUTABLE';
     USING SQL $$
-    SELECT date_trunc("unit", "dt")
+    SELECT CASE WHEN "unit" in ('microseconds', 'milliseconds')
+        THEN date_trunc("unit", "dt")
+        WHEN "unit" in ('seconds', 'minutes', 'hours')
+        THEN date_trunc(substring("unit" for length("unit")-1), "dt")
```

Comment:
As far as I can tell Postgres accepts both singular and plural units, so there's no need for truncating the "s" from the end of the unit name.

Reply:
Yes, you're right. Unfortunately, I don't see it documented anywhere :(

Reply:
Oh, the docs only list "microseconds" and "milliseconds" as plurals, the rest are singular. That's annoying. Actually, at this point if I had to choose I'd probably be more inclined to pick singular unit format. It makes no difference for `duration`, but `relativedelta` would benefit from using singular units because it's a little annoying to use "centuries" and guess the correct plural of "millennium" ("millennia" is correct Latin plural, but "millenniums" is also acceptable in English).